Bug: 대기열 서버 시작시 redis 스트림을 초기화하지 못하는 오류

## 🐛 어떤 버그인가요?
> 대기열 서버 시작시 redis 스트림을 초기화하지 못하는 오류

## 🔍️ 어떤 상황에서 발생한 버그인가요?
> 
```
Caused by: io.lettuce.core.RedisCommandExecutionException: NOGROUP No such key 'DISPATCH' or consumer group 'DISPATCH:waiting-1' in XREADGROUP with GROUP option
        at io.lettuce.core.internal.ExceptionFactory.createExecutionException(ExceptionFactory.java:151) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.lettuce.core.internal.ExceptionFactory.createExecutionException(ExceptionFactory.java:120) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.lettuce.core.protocol.AsyncCommand.completeResult(AsyncCommand.java:124) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.lettuce.core.protocol.AsyncCommand.complete(AsyncCommand.java:115) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.lettuce.core.protocol.CommandWrapper.complete(CommandWrapper.java:67) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.lettuce.core.protocol.CommandHandler.complete(CommandHandler.java:749) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.lettuce.core.protocol.CommandHandler.decode(CommandHandler.java:684) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.lettuce.core.protocol.CommandHandler.channelRead(CommandHandler.java:601) ~[lettuce-core-6.4.2.RELEASE.jar!/:6.4.2.RELEASE/f4dfb40]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998) ~[netty-common-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.119.Final.jar!/:4.1.119.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.119.Final.jar!/:4.1.119.Final]
        ... 1 common frames omitted
```

## 💡 예상 결과
> 예상했던 정상적인 결과가 어떤 것이었는지 설명해주세요

## 🔗 Related Issues 
[이슈 제목](이슈 링크)

## 🏷️ 참고할만한 자료(선택)

Closes #{이슈번호}

--- a/service/common/src/main/java/org/codenbug/common/util/CookieUtil.java
+++ b/service/common/src/main/java/org/codenbug/common/util/CookieUtil.java
@@ -36,7 +36,7 @@ public class CookieUtil {
 		Cookie cookie = new Cookie(ACCESS_TOKEN_COOKIE_NAME, token);
 		cookie.setMaxAge((int)(accessTokenExpiration / 1000)); // milliseconds to seconds
 		cookie.setPath(COOKIE_PATH);
-		// cookie.setDomain(cookieDomain);
+		cookie.setDomain(cookieDomain);
 		cookie.setHttpOnly(true);
 
 		// 개발 환경 확인 (localhost에서는 secure=false로 설정)
@@ -57,7 +57,7 @@ public class CookieUtil {
 		Cookie cookie = new Cookie(REFRESH_TOKEN_COOKIE_NAME, token);
 		cookie.setMaxAge((int)(refreshTokenExpiration / 1000)); // milliseconds to seconds
 		cookie.setPath(COOKIE_PATH);
-		// cookie.setDomain(cookieDomain);
+		cookie.setDomain(cookieDomain);
 		cookie.setHttpOnly(true);
 
 		// 개발 환경 확인 (localhost에서는 secure=false로 설정)
@@ -77,7 +77,7 @@ public class CookieUtil {
 		Cookie cookie = new Cookie(ACCESS_TOKEN_COOKIE_NAME, null);
 		cookie.setMaxAge(0);
 		cookie.setPath(COOKIE_PATH);
-		// cookie.setDomain(cookieDomain);
+		cookie.setDomain(cookieDomain);
 		cookie.setHttpOnly(true);
 
 		// 개발 환경 확인
@@ -97,7 +97,7 @@ public class CookieUtil {
 		Cookie cookie = new Cookie(REFRESH_TOKEN_COOKIE_NAME, null);
 		cookie.setMaxAge(0);
 		cookie.setPath(COOKIE_PATH);
-		// cookie.setDomain(cookieDomain);
+		cookie.setDomain(cookieDomain);
 		cookie.setHttpOnly(true);
 
 		// 개발 환경 확인

--- a/service/main-server/src/main/java/org/codeNbug/mainserver/domain/event/repository/QueryDslCommonEventRepository.java
+++ b/service/main-server/src/main/java/org/codeNbug/mainserver/domain/event/repository/QueryDslCommonEventRepository.java
@@ -6,6 +6,7 @@ import org.codeNbug.mainserver.domain.event.dto.request.EventListFilter;
 import org.codeNbug.mainserver.domain.event.entity.CommonEventRepository;
 import org.codeNbug.mainserver.domain.event.entity.Event;
 import org.codeNbug.mainserver.domain.event.entity.QEvent;
+import org.codeNbug.mainserver.domain.seat.entity.QSeat;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageImpl;
 import org.springframework.data.domain.Pageable;
@@ -82,12 +83,13 @@ public class QueryDslCommonEventRepository implements CommonEventRepository {
 
 	@Override
 	public Integer countAvailableSeat(Long id) {
-		return jpaQueryFactory
-			.select(QEvent.event.seatLayout.seats.size())
-			.from(QEvent.event)
-			.where(QEvent.event.eventId.eq(id)
-				.and(QEvent.event.seatLayout.seats.any().available.isTrue())
-				.and(filterDeletedFalseExpression()))
-			.fetchOne();
+		return Math.toIntExact(jpaQueryFactory
+			.select(QSeat.seat.count()) // 단순히 개수를 셀 것이므로 seat.count() 또는 다른 표현도 가능합니다.
+			.from(QSeat.seat)
+			.where(
+				QSeat.seat.event.eventId.eq(id), // 기존 이벤트 ID 조건
+				QSeat.seat.available.isTrue()    // available이 true인 좌석만 선택하는 조건 추가!
+			)
+			.fetchOne());
 	}
 }

--- a/service/main-server/src/main/resources/static/payment/payment.html
+++ b/service/main-server/src/main/resources/static/payment/payment.html
@@ -1,0 +1,52 @@
+<!DOCTYPE html>
+
+<html lang="ko">
+
+<head>
+
+    <meta charset="UTF-8">
+
+    <title>Toss Payments Test</title>
+
+    <script src="https://js.tosspayments.com/v1/payment"></script>
+
+</head>
+
+<body>
+
+<button id="payment-button">결제하기</button>
+
+
+<script>
+
+    const clientKey = 'test_ck_Z61JOxRQVEY6lZeGL4zgVW0X9bAq'; // 내 테스트 Client Key로 바꿔주세요
+
+
+    const tossPayments = TossPayments(clientKey);
+
+
+    document.getElementById('payment-button').addEventListener('click', () => {
+
+        tossPayments.requestPayment('카드', {
+
+            amount: 1000,
+
+            orderId: 'order-' + Date.now(),
+
+            orderName: '테스트 결제',
+
+            successUrl: 'https://your-success-url.com', // 성공 시 이동할 주소
+
+            failUrl: 'https://your-fail-url.com',       // 실패 시 이동할 주소
+
+            customerName: '테스트 유저'
+
+        });
+
+    });
+
+</script>
+
+</body>
+
+</html>

--- a/service/main-server/src/test/java/org/codeNbug/mainserver/domain/event/repository/QueryDslCommonEventRepositoryTest.java
+++ b/service/main-server/src/test/java/org/codeNbug/mainserver/domain/event/repository/QueryDslCommonEventRepositoryTest.java
@@ -1,0 +1,138 @@
+package org.codeNbug.mainserver.domain.event.repository;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.codeNbug.mainserver.domain.event.entity.Event;
+import org.codeNbug.mainserver.domain.event.entity.EventCategoryEnum;
+import org.codeNbug.mainserver.domain.event.entity.EventInformation;
+import org.codeNbug.mainserver.domain.event.entity.EventStatusEnum;
+import org.codeNbug.mainserver.domain.manager.repository.EventRepository;
+import org.codeNbug.mainserver.domain.seat.entity.Seat;
+import org.codeNbug.mainserver.domain.seat.entity.SeatGrade;
+import org.codeNbug.mainserver.domain.seat.entity.SeatGradeEnum;
+import org.codeNbug.mainserver.domain.seat.entity.SeatLayout;
+import org.codeNbug.mainserver.domain.seat.repository.SeatGradeRepository;
+import org.codeNbug.mainserver.domain.seat.repository.SeatLayoutRepository;
+import org.codeNbug.mainserver.domain.seat.repository.SeatRepository;
+import org.codeNbug.mainserver.global.config.QueryDslConfig;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Import;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@DataJpaTest
+@Testcontainers
+@ComponentScan(basePackages = {"org.codeNbug.mainserver.domain.event.repository"})
+@EntityScan(basePackages = {"org.codeNbug.mainserver.domain", "org.codenbug.user.domain"})
+@Import(QueryDslConfig.class)
+class QueryDslCommonEventRepositoryTest {
+
+	@Container
+	@ServiceConnection
+	static MySQLContainer<?> mysql =
+		new MySQLContainer<>("mysql:8.0.34")
+			.withDatabaseName("ticketoneTest")
+			.withUsername("test")
+			.withPassword("test");
+
+	@Autowired
+	private EventRepository eventRepository;
+
+	@Autowired
+	private SeatLayoutRepository seatLayoutRepository;
+
+	@Autowired
+	private SeatGradeRepository seatGradeRepository;
+
+	@Autowired
+	private SeatRepository seatRepository;
+
+	@Autowired
+	private QueryDslCommonEventRepository queryDslCommonEventRepository;
+
+	@Test
+	@DisplayName("예매 가능한 좌석 조회 테스트")
+	void countAvailableTest() {
+
+		// 임의의 행사를 db에 저장
+		EventInformation eventInformation = EventInformation.builder()
+			.title("Test Event")
+			.description("Test Description")
+			.location("Test Location")
+			.hallName("Test Hall")
+			.eventStart(LocalDateTime.now().plusDays(10))
+			.eventEnd(LocalDateTime.now().plusDays(11))
+			.seatCount(10)
+			.build();
+
+		Event event = new Event(
+			EventCategoryEnum.CONCERT,
+			eventInformation,
+			LocalDateTime.now(),
+			LocalDateTime.now().plusDays(5),
+			0,
+			LocalDateTime.now(),
+			LocalDateTime.now(),
+			EventStatusEnum.OPEN,
+			true,
+			false,
+			null
+		);
+
+		Event savedEvent = eventRepository.save(event);
+
+		// Create SeatLayout
+		SeatLayout seatLayout = SeatLayout.builder()
+			.layout("Test Layout")
+			.event(savedEvent)
+			.build();
+
+		SeatLayout savedSeatLayout = seatLayoutRepository.save(seatLayout);
+		savedEvent.setSeatLayout(savedSeatLayout);
+		eventRepository.save(savedEvent);
+
+		// Create SeatGrade
+		SeatGrade seatGrade = SeatGrade.builder()
+			.grade(SeatGradeEnum.VIP)
+			.amount(10000)
+			.event(savedEvent)
+			.build();
+
+		SeatGrade savedSeatGrade = seatGradeRepository.save(seatGrade);
+
+		// Create 10 seats
+		for (int i = 0; i < 10; i++) {
+			Seat seat = Seat.builder()
+				.location("Seat " + i)
+				.available(true)
+				.grade(savedSeatGrade)
+				.layout(savedSeatLayout)
+				.event(savedEvent)
+				.build();
+
+			seatRepository.save(seat);
+		}
+
+		// 저장한 행사의 seat 2개를 골라 available을 false로 설정
+		List<Seat> seats = seatRepository.findAll();
+		seats.get(0).setAvailable(false);
+		seats.get(1).setAvailable(false);
+		seatRepository.saveAll(seats);
+
+		// 예매 가능 좌석 조회 메소드 실행
+		Integer availableSeatCount = queryDslCommonEventRepository.countAvailableSeat(savedEvent.getEventId());
+
+		// 총 행사의 seat - 2가 나와야 함
+		assertEquals(8, availableSeatCount);
+	}
+}

